### ISSUE TITLE
First release?

### ISSUE THREAD

**Issue by kloczek** [nothings/stb#1352]
Is it possible to make the first release out of the code alredy accumulated in git repo?
(+ maybe few PRs)

**Comment by nothings** [comment]
Can you rephrase that? I'm not sure what you're asking.

**Comment by kloczek** [comment]
stb has not been released even one time.
https://github.com/nothings/stb/tags is empty.

**Comment by nothings** [comment]
It has been released plenty of times; the public repository is the release.

As to _tagged_ releases, see the discussion in https://github.com/nothings/stb/issues/359, in which nobody was able to suggest a position that was compatible with the notion of multiple independent libraries in a single github project.

**Comment by kloczek** [comment]
OK so very simple question: what is tha latest stb version and where I can find info aboyt versio? 🤔

**Comment by nothings** [comment]
there is no version for stb as a whole. each library has its own version number, which can be found in the file itself at the top. the versions are also listed in the table on the front page under the directory listing.

**Comment by kloczek** [comment]
So why did you wprote that stb has been releaset plenty of times if it wasn't released with exaxct version?
Versioning is important because it allows add build dependencies.
Please make first regular release like 1.0.0.
And/or is is any problem with make such first release? 🤔

**Comment by trapexit** [comment]
What build dependencies? The point of single header libraries is to embed them into your project directly. And as mentioned each library is versioned. There is no collective version for the repo nor should there be. It wouldn't make sense. These are largely independent libraries packaged together physically for convenience.

**Comment by kloczek** [comment]
You may be don't know that but stb is packaged in many distributions.
Layer abowe on which you are working (packaging layer) people defines build dependencies.
Also many distributions monitors ftp/http/https URLs and gitlab/github to obtain informations that maintainer just decided to make new release and probably it would be good to have look on exact package to update it.
Projects without version makes all that waaay harder to track.

**Comment by trapexit** [comment]
It may be packaged in distributions but, for example, if you look at Debian they had to convert it into something it isn't nor intended to be.

https://salsa.debian.org/yangfl-guest/stb/-/blob/master/debian/patches/0001-Makefile-refactor.patch

It is in effect its own project. Taking disperse projects and combining them into one.

And it's really only used by wine on Debian.

Each library in stb is it's own thing and as mentioned versioned independently outside of the source control system. If there is a system watching anything it should watch the lastest individual files and not the repo.

**Comment by nothings** [comment]
Please take further discussion to #359.
